### ISSUE TITLE
Dask-cuda-worker nanny process creates gpu context

### ISSUE THREAD

**Issue by ayushdg** [rapidsai/dask-cuda#137]
Process:
```
$ dask-scheduler
distributed.scheduler - INFO - -----------------------------------------------
distributed.scheduler - INFO - Local Directory:    /tmp/scheduler-pt28rj10
distributed.scheduler - INFO - -----------------------------------------------
distributed.scheduler - INFO - Clear task state
distributed.scheduler - INFO -   Scheduler at:   tcp://172.0.0.1:8786
distributed.scheduler - INFO -   dashboard at:                     :8787
```

```
$ dask-cuda-worker tcp://172.0.0.1:8786
distributed.nanny - INFO -         Start Nanny at: 'tcp://127.0.0.1:41197'
distributed.nanny - INFO -         Start Nanny at: 'tcp://127.0.0.1:39161'
distributed.nanny - INFO -         Start Nanny at: 'tcp://127.0.0.1:43223'
distributed.nanny - INFO -         Start Nanny at: 'tcp://127.0.0.1:34056'
distributed.diskutils - INFO - Found stale lock file and directory '/dask-worker-space/worker-an364ax5', purging
distributed.diskutils - INFO - Found stale lock file and directory '/dask-worker-space/worker-yitxqoaq', purging
distributed.diskutils - INFO - Found stale lock file and directory '/dask-worker-space/worker-1_ru1p_8', purging
distributed.diskutils - INFO - Found stale lock file and directory '/dask-worker-space/worker-l2d_as3n', purging
distributed.preloading - INFO - Import preload module: dask_cuda.initialize_context
distributed.preloading - INFO - Import preload module: dask_cuda.initialize_context
distributed.worker - INFO -       Start worker at:   tcp://127.0.0.1:43405
distributed.worker - INFO -          Listening to:   tcp://127.0.0.1:43405
distributed.worker - INFO -          dashboard at:         127.0.0.1:45235
distributed.worker - INFO - Waiting to connect to:    tcp://127.0.0.1:8786
distributed.worker - INFO - -------------------------------------------------
distributed.worker - INFO -               Threads:                          1
distributed.worker - INFO -                Memory:                    6.32 GB
distributed.worker - INFO -       Local Directory: /dask-worker-space/worker-ldfu7d33
distributed.worker - INFO - -------------------------------------------------
distributed.worker - INFO -       Start worker at:   tcp://127.0.0.1:46117
distributed.worker - INFO -          Listening to:   tcp://127.0.0.1:46117
distributed.worker - INFO -          dashboard at:         127.0.0.1:46306
distributed.worker - INFO - Waiting to connect to:    tcp://127.0.0.1:8786
distributed.worker - INFO - -------------------------------------------------
distributed.worker - INFO -               Threads:                          1
distributed.worker - INFO -                Memory:                    6.32 GB
distributed.worker - INFO -       Local Directory: /dask-worker-space/worker-3u5yjvh2
distributed.worker - INFO - -------------------------------------------------
distributed.preloading - INFO - Import preload module: dask_cuda.initialize_context
distributed.worker - INFO -         Registered to:    tcp://127.0.0.1:8786
distributed.worker - INFO - -------------------------------------------------
distributed.preloading - INFO - Import preload module: dask_cuda.initialize_context
distributed.core - INFO - Starting established connection
distributed.worker - INFO -         Registered to:    tcp://127.0.0.1:8786
distributed.worker - INFO - -------------------------------------------------
distributed.core - INFO - Starting established connection
distributed.worker - INFO -       Start worker at:   tcp://127.0.0.1:38330
distributed.worker - INFO -          Listening to:   tcp://127.0.0.1:38330
distributed.worker - INFO -          dashboard at:         127.0.0.1:37211
distributed.worker - INFO - Waiting to connect to:    tcp://127.0.0.1:8786
distributed.worker - INFO - -------------------------------------------------
distributed.worker - INFO -               Threads:                          1
distributed.worker - INFO -                Memory:                    6.32 GB
distributed.worker - INFO -       Local Directory: /dask-worker-space/worker-60ez49uj
distributed.worker - INFO - -------------------------------------------------
distributed.worker - INFO -       Start worker at:   tcp://127.0.0.1:46114
distributed.worker - INFO -          Listening to:   tcp://127.0.0.1:46114
distributed.worker - INFO -          dashboard at:         127.0.0.1:34688
distributed.worker - INFO - Waiting to connect to:    tcp://127.0.0.1:8786
distributed.worker - INFO - -------------------------------------------------
distributed.worker - INFO -               Threads:                          1
distributed.worker - INFO -                Memory:                    6.32 GB
distributed.worker - INFO -       Local Directory: /dask-worker-space/worker-yud8mbgi
distributed.worker - INFO - -------------------------------------------------
distributed.worker - INFO -         Registered to:    tcp://127.0.0.1:8786
distributed.worker - INFO - -------------------------------------------------
distributed.core - INFO - Starting established connection
distributed.worker - INFO -         Registered to:    tcp://127.0.0.1:8786
distributed.worker - INFO - -------------------------------------------------
distributed.core - INFO - Starting established connection
```

Output for Nvidia-smi
```
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 410.104      Driver Version: 410.104      CUDA Version: 10.0     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla T4            Off  | 00000000:3B:00.0 Off |                    0 |
| N/A   45C    P0    28W /  70W |    248MiB / 15079MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   1  Tesla T4            Off  | 00000000:5E:00.0 Off |                    0 |
| N/A   46C    P0    28W /  70W |    248MiB / 15079MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   2  Tesla T4            Off  | 00000000:AF:00.0 Off |                    0 |
| N/A   45C    P0    28W /  70W |    248MiB / 15079MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   3  Tesla T4            Off  | 00000000:D8:00.0 Off |                    0 |
| N/A   44C    P0    28W /  70W |    248MiB / 15079MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0     43392      C   /opt/conda/envs/rapids/bin/python            119MiB |
|    0     43488      C   /opt/conda/envs/rapids/bin/python            119MiB |
|    1     43392      C   /opt/conda/envs/rapids/bin/python            119MiB |
|    1     43490      C   /opt/conda/envs/rapids/bin/python            119MiB |
|    2     43392      C   /opt/conda/envs/rapids/bin/python            119MiB |
|    2     43492      C   /opt/conda/envs/rapids/bin/python            119MiB |
|    3     43392      C   /opt/conda/envs/rapids/bin/python            119MiB |
|    3     43494      C   /opt/conda/envs/rapids/bin/python            119MiB |
+-----------------------------------------------------------------------------+
```

Previously there would be 4 processes, one for each worker showing up in `Nvidia-smi`. Is this expected?

**Comment by pentschev** [comment]
@ayushdg thanks for the report.

It's not ideal, no. But the main process creates a context and the forked workers create another. I already started working on a fix in #105 but it's a bit more complicated and I had no time to finish yet. This is a duplicate of https://github.com/rapidsai/dask-cuda/issues/104, thus I'm closing it.

**Comment by ayushdg** [comment]
Thanks @pentschev for the response! :D
